docs(ux-writing): Adds conversation design and chatbot guidelines.

edonehoo opened the pull request:
Closes #4001 
Preview link: https://patternfly-org-pr-4206-site.surge.sh/patternfly-ai/conversation-design

patternfly-build commented:
Preview: https://patternfly-org-pr-4206-site.surge.sh

edonehoo commented:
Edit: let's release this alongside the chatbot, I'll make updates to align them better when the release is closer. We can also revisit these docs once published and remove anything that's extraneous/unnecessary

andrew-ronaldson commented:
Not a PR note but noticed the content isn't wrapping on mobile. @srambach any thoughts?

![image](https://github.com/user-attachments/assets/f75c6379-81b7-4df6-9ee1-4a074ba4fe90)

edonehoo commented:
Update - Adjusted this a bit to align with the chatbot extension, and I'm planning to move some top level info into an "about chatbot" kind of page via https://github.com/patternfly/virtual-assistant/issues/182

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/AI/conversation-design/chatbots.md:
```diff
@@ -0,0 +1,64 @@
+
+## Writing for chatbots 
+
+When chatbots are design to meet the needs of your users, they can improve the overall UX of your product, because they are convenient, efficient, and persistent. 
```
```suggestion
When chatbots are designed to meet the needs of your users, they can improve the overall UX of your product, because they are convenient, efficient, and persistent. 
```

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/AI/conversation-design/chatbots.md:
```diff
@@ -0,0 +1,64 @@
+<GalleryItem>
+<Card isFullHeight isLarge >
+<CardTitle>AI-based chatbots</CardTitle>
+    <CardBody>Bots that use technologies like natural language processing and generative AI to parse user input and decide how to response. Often makes for a more natural, free flowing conversation, but relies on a trustworthy AI model.</CardBody>
```
```suggestion
    <CardBody>Bots that use technologies like natural language processing and generative AI to parse user input and decide how to respond. Often makes for a more natural, free flowing conversation, but relies on a trustworthy AI model.</CardBody>
```

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/AI/conversation-design/chatbots.md:
```diff
@@ -0,0 +1,64 @@
+
+## When should I use a chatbot?
+
+It is important to only use chatbots when they add value to the user experience. Do not use chatbots simply for the sake of novelty. Your users aren't likely to be excited  to interact with a new chatbot feature, unless it directly helps them.
```
```suggestion
It is important to only use chatbots when they add value to the user experience. Do not use chatbots simply for the sake of novelty. Your users aren't likely to be excited to interact with a new chatbot feature, unless it directly helps them.
```

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/AI/conversation-design/conversation-design.md:
```diff
@@ -0,0 +1,47 @@
+
+## Best practices 
+
+These best practices for conversation design will help ensure that your users needs are met and that their experience with your chatbot is positive and painless. 
```
```suggestion
These best practices for conversation design will help ensure that your users' needs are met and that their experience with your chatbot is positive and painless. 
```

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/AI/conversation-design/conversation-design.md:
```diff
@@ -0,0 +1,47 @@
+- Avoid open ended questions. If you ask any, make them very simple, increase pause time between messages, and tell users when they should type. 
+- Be direct. Limit choices for users as much as possible and provide examples when you can.
+- Frame questions in terms of value for the user.
+- Unlike traditional content design, *don't lead with the benefit.* End with the benefit to be sure that its read and acted upon. Users should be able to complete an action solely based on a chatbot's last sentence. 
```
```suggestion
- Unlike traditional content design, *don't lead with the benefit.* End with the benefit to be sure that it is read and acted upon. Users should be able to complete an action solely based on a chatbot's last sentence. 
```

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/PatternFly-AI/conversation-design/conversation-design.md:
```diff
@@ -0,0 +1,75 @@
+
+## Writing for chatbots
+
+When chatbots are design to meet the needs of your users, they can improve the overall UX of your product, because they are convenient, efficient, and persistent. 
```
```suggestion
When chatbots are designed to meet the needs of your users, they can improve the overall UX of your product. They are convenient, efficient, and persistent. 
```

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/PatternFly-AI/conversation-design/conversation-design.md:
```diff
@@ -0,0 +1,75 @@
+
+When chatbots are design to meet the needs of your users, they can improve the overall UX of your product, because they are convenient, efficient, and persistent. 
+
+[Our virtual assistant extension](/patterfnfly-ai/chatbot/about-virtual-assistant) utilizes PatternFly components to create a foundation for an AI-based chatbot, with additional customization options.
```
```suggestion
[Our virtual assistant extension](/patternfly-ai/chatbot/about-virtual-assistant) utilizes PatternFly components to create a foundation for an AI-based chatbot, with additional customization options.
```

rebeccaalpert replied:
This link doesn't exist either though; I think we are using patternfly-ai/chatbot/about-chatbot?

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/PatternFly-AI/conversation-design/conversation-design.md:
```diff
@@ -0,0 +1,75 @@
+
+- Use labels and other visual styling cues to clearly identify AI features. 
+- Add necessary legal disclosures where you can, like in the chatbot footer. 
+- Display an indicator when the bot is "thinking" or "typing", so that users know to expect feedback.
```
```suggestion
- Display an indicator when the bot is "thinking" or "typing," so that users know to expect feedback.
```

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/PatternFly-AI/conversation-design/conversation-design.md:
```diff
@@ -0,0 +1,75 @@
+
+Good conversation design combines content strategy, writing, and design to ensure that the development and use of chatbots is centered on real user needs. This includes making sure that all text related to the conversational UI supports easy engagement with chatbots, while staying aligned with brand standards.
+
+When designing AI conversations alongside PatternFly-based projects, it is important to align with our established [brand voice and tone](/ux-writing/brand-voice-and-tone), as well as our [ethical guidelines for AI.](/patternfly-ai/ai-guidelines)
```
"ethical guidelines for AI" link is broken.

edonehoo replied:
this page should be added by https://github.com/patternfly/patternfly-org/pull/4175! will probably be worth a quick double check on links once all the AI docs are merged to make sure they work

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/AI/conversation-design/chatbots.md:
```diff
@@ -0,0 +1,64 @@
+<Card isFullHeight isLarge>
+<CardTitle>Rule-based chatbots</CardTitle>
+    <CardBody>Bots that are based on a set of predefined rules. These use a planned, guided dialog, and have a script that the user must stay on. The script is written by a human, including any microcopy, rules, prompts, and responses.</CardBody>
+    <CardFooter> <Button component="a" href="https://www.patternfly.org/ux-writing/chatbots" variant="link" icon={<ArrowRightIcon />} iconPosition="end"> View chatbot writing guidelines  </Button></CardFooter>
```
Will this link exist?

edonehoo replied:
oops, I actually meant to delete this page!

rebeccaalpert commented on packages/documentation-site/patternfly-docs/content/AI/conversation-design/conversation-design.md:
```diff
@@ -0,0 +1,47 @@
+
+**Conversation design** is a subset of UX that's focused on the design of conversation between humans and technology, most often through chatbots or voicebots. The goal of conversation design is to create an interactive experience that resembles human-to-human conversation as much as possible.
+
+It is important to align conversation design with the established [PatternFly brand voice and tone](/ux-writing/brand-voice-and-tone), as well as our [AI guidelines](/ai/ai-guidelines) and [principles.](/ai/about-ai)
```
Will these links exist?

rebeccaalpert reviewed (changes requested):
This looks good to me overall! I can only see the one page (https://patternfly-org-pr-4206-site.surge.sh/patternfly-ai/conversation-design), so that's all I checked visually. I noted some suspected typos and added some questions/comments about some links.

nicolethoen commented on packages/documentation-site/patternfly-docs/content/PatternFly-AI/conversation-design/conversation-design.md:
```diff
@@ -0,0 +1,75 @@
+---
+id: Conversation design
+section: PatternFly-AI
```
Which sections do we want this and the chatbots.md files to appear in the nav? Should they be under patternfly-ai/conversation-design as different tabs?

if so, i think you should be able to do that by using 
```
id: (page title)
section: PatternFly-AI
source: (page title again? ... really it's whatever you want to be on the tab)
```

nicolethoen reviewed (commented):
Some pages are hidden

edonehoo commented:
ty for the reviews 😁 

so, my bad for the confusion--I accidentally forgot to remove a couple files from this pr! I only want to add patternfly-ai/conversation-design via this PR. Any info that I removed from earlier pr iterations, which is more chatbot-specific, I'll add back to the chatbots subsection as part of this pr https://github.com/patternfly/virtual-assistant/issues/182.

All of the various PRs would then combine to resemble this in the nav:

PatternFly AI
- About AI
- AI guidelines 
- Conversation design
- Chatbot
    - About chatbot (where I'll move some of the info I just deleted from this pr)
    - Continue with chatbot specific docs etc

But please LMK if this plan doesn't make sense or if y'all think we should go about it differently @rebeccaalpert @nicolethoen

andrew-ronaldson reviewed (approved):
I'm speechless!

rebeccaalpert reviewed (approved):
Looks awesome! We'll just have to update the links to the chatbot here later since the structure was changed in your other PR in the virtual-assistant repo. I don't think these links will be correct given the changes there.
